common/src/main/java/com/wynntils/functions/BombFunctions.java

```diff
@@ -0,0 +1,134 @@
+                    new Argument<>("index", Integer.class, null),
+                    new Argument<>("group", Boolean.class, null),
+                    new Argument<>("sortOrder", String.class, null)));
+        }
+
+        @Override
+        public T getValue(FunctionArguments arguments) {
+            int index = arguments.getArgument("index").getIntegerValue();
+            boolean group = arguments.getArgument("group").getBooleanValue();
+            BombSortOrder sortOrder =
+                    BombSortOrder.fromString(arguments.getArgument("sortOrder").getStringValue());
+            List<BombInfo> bombInfo =
+                    Models.Bomb.getBombBellStream(group, sortOrder).toList();
+
+            return (!bombInfo.isEmpty() && index >= 0 && index <= bombInfo.size())
+                    ? processInfo(bombInfo.get(index))
+                    : invalidValue();
+        }
+
+        @Override
+        public String getArgumentDescription(String argumentName) {
+            return I18n.get("function.wynntils.bombFunctionBase.argument." + argumentName);
+        }
+
+        public abstract T processInfo(BombInfo info);
+
+        public abstract T invalidValue();
+    }
+
+    public static class BombFormattedStringFunction extends BombFunctionBase<String> {
+        @Override
+        public String processInfo(BombInfo info) {
+            return info.asString();
+        }
+
+        @Override
+        public String invalidValue() {
+            return "";
+        }
+    }
+
+    public static class BombTypeFunction extends BombFunctionBase<String> {
+        @Override
+        public String processInfo(BombInfo info) {
+            return info.bomb().getDisplayName();
+        }
+
+        @Override
+        public String invalidValue() {
+            return "";
+        }
+    }
+
+    public static class BombWorldFunction extends BombFunctionBase<String> {
+        @Override
+        public String processInfo(BombInfo info) {
+            return info.server();
+        }
+
+        @Override
+        public String invalidValue() {
+            return "";
+        }
+    }
+
+    public static class BombStartTimeFunction extends BombFunctionBase<Time> {
+        @Override
+        public Time processInfo(BombInfo info) {
+            return Time.of(info.startTime());
+        }
+
+        @Override
+        public Time invalidValue() {
+            return Time.of(-1);
+        }
+    }
+
+    public static class BombLengthFunction extends BombFunctionBase<Float> {
+        @Override
+        public Float processInfo(BombInfo info) {
+            return info.length();
+        }
+
+        @Override
+        public Float invalidValue() {
+            return -1f;
+        }
+    }
+
+    public static class BombRemainingTimeFunction extends BombFunctionBase<Long> {
+        @Override
+        public Long processInfo(BombInfo info) {
+            return info.getRemainingLong();
```

Comment:
After reading the code, I suggest you convert this to BombEndTimeFunction. You can update BombInfo to expose an `endTime()`.